scout/server/blueprints/variants/controllers.py

```diff
@@ -805,28 +805,66 @@ def populate_sv_filters_form(store, institute_obj, case_obj, category, request_o
     # populate available panel choices
     form.gene_panels.choices = gene_panel_choices(institute_obj, case_obj)
 
-    # check if supplied gene symbols exist
+    form = update_form_hgnc_symbols(store, case_obj, form)
+
+    return form
+
+
+def update_form_hgnc_symbols(store, case_obj, form):
+    """Update variants filter form with HGNC symbols from HPO, and check if any non-clinical genes for the case were
+    requested. If so, flash a warning to the user.
+
+    Accepts:
+        store(adapter.MongoAdapter)
+        case_obj(dict)
+        form(FiltersForm)
+
+    Returns:
+        form(FiltersForm)
+    """
+
     hgnc_symbols = []
     non_clinical_symbols = []
     not_found_symbols = []
     not_found_ids = []
+    updated_hgnc_symbols = []
+
+    # retrieve current symbols from form
     if (form.hgnc_symbols.data) and len(form.hgnc_symbols.data) > 0:
-        is_clinical = form.data.get("variant_type", "clinical") == "clinical"
-        clinical_symbols = store.clinical_symbols(case_obj) if is_clinical else None
+        # if symbols are numeric HGNC id, translate to current symbols
         for hgnc_symbol in form.hgnc_symbols.data:
             if hgnc_symbol.isdigit():
                 hgnc_gene = store.hgnc_gene(int(hgnc_symbol), case_obj["genome_build"])
                 if hgnc_gene is None:
                     not_found_ids.append(hgnc_symbol)
                 else:
                     hgnc_symbols.append(hgnc_gene["hgnc_symbol"])
-            elif sum(1 for i in store.hgnc_genes(hgnc_symbol)) == 0:
-                not_found_symbols.append(hgnc_symbol)
-            elif is_clinical and (hgnc_symbol not in clinical_symbols):
-                non_clinical_symbols.append(hgnc_symbol)
             else:
                 hgnc_symbols.append(hgnc_symbol)
 
+    # add HPO genes to list, if they were missing
+    if "hpo" in form.data["gene_panels"]:
+        hpo_symbols = list(
+            set(term_obj["hgnc_symbol"] for term_obj in case_obj["dynamic_gene_list"])
+        )
+
+        current_symbols = set(hgnc_symbols)
+        current_symbols.update(hpo_symbols)
+
+        hgnc_symbols = list(current_symbols)
+
+    # check if supplied gene symbols exist and are clinical
+    is_clinical = form.data.get("variant_type", "clinical") == "clinical"
+    clinical_symbols = store.clinical_symbols(case_obj) if is_clinical else None
+    for hgnc_symbol in hgnc_symbols:
+        if sum(1 for i in store.hgnc_genes(hgnc_symbol)) == 0:
+            not_found_symbols.append(hgnc_symbol)
+        elif is_clinical and (hgnc_symbol not in clinical_symbols):
+            non_clinical_symbols.append(hgnc_symbol)
+        else:
+            updated_hgnc_symbols.append(hgnc_symbol)
```

Comment:
It is not - probably some uncertainty on wether to change the user input or not that shone through. I currently feel it should be used to update the search filed, and made it so, but discussion is welcome!

Reply:
Agreed: the intention though would be to flash it once each time that gene is entered - or HPO panel is clicked - but just update that gene string anyway. I get a nagging feeling someone may use that search string field for copying and pasting to other places - and then it may well be up-to-date?